client/rest-high-level/src/main/java/org/opensearch/client/SearchPipelinesClient.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.search.DeleteSearchPipelineRequest;
+import org.opensearch.action.search.GetSearchPipelineRequest;
+import org.opensearch.action.search.GetSearchPipelineResponse;
+import org.opensearch.action.search.PutSearchPipelineRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import static java.util.Collections.emptySet;
+
+public final class SearchPipelinesClient {
```

Comment:
@msfroh I would suggest to settle on same naming conventions, use sporadically plural `SearchPipelinesClient` and singular `SearchPipelineService` forms, would be good to have it just concise everyone.

Reply:
You're right. I looked through the list of changed files, and I still see the plural form used for this client and the associated YAML tests. I'll update those to use the singular form.

Right now, `SearchPipelinesInfo` doesn't have information about any search pipelines (singular or plural), but rather keeps track of the available search pipeline processors, so that we can validate on pipeline creation that all nodes have all processors mentioned in the pipeline. Given that it describes information at the node level, I don't think it would ever hold information about search pipelines themselves (whether a single pipeline or all pipelines), since the pipelines themselves are defined at the cluster level. So, after that roundabout thought process, I think I'll rename it from `SearchPipelinesInfo` to `SearchPipelineInfo`, since it's ultimately carrying information about the node-level `SearchPipelineService`.